modules/process/manager.go

```diff
@@ -49,34 +70,129 @@ func GetManager() *Manager {
 	managerInit.Do(func() {
 		manager = &Manager{
 			processes: make(map[int64]*Process),
+			next:      1,
+			low:       1,
 		}
 	})
 	return manager
 }
 
-// Add a process to the ProcessManager and returns its PID.
-func (pm *Manager) Add(description string, cancel context.CancelFunc) int64 {
+// AddContext create a new context and add it as a process. The CancelFunc must always be called even if the context is Done()
+func (pm *Manager) AddContext(parent context.Context, description string) (context.Context, context.CancelFunc) {
```

Comment:
I think we might need to provide a separate pure cancel function here in addition to the cancel and remove fn, or perhaps provide the ability to pass in a cancel() that runs between the process being cancelled and the process being removed.

It appears that the cancellation of process contexts doesn't always directly lead to the killing of the process - hence a second step may be necessary.